From 26148417fab419a0c7f301fb8f2be015324d5374 Mon Sep 17 00:00:00 2001
From: Steffen Trumtrar <s.trumtrar@pengutronix.de>
Date: Fri, 30 Jun 2017 16:53:17 +0200
Subject: libdt: support finding devices by partuuid

Signed-off-by: Steffen Trumtrar <s.trumtrar@pengutronix.de>

diff --git a/src/libdt.c b/src/libdt.c
index 3adeed2..2bc6cc1 100644
--- a/src/libdt.c
+++ b/src/libdt.c
@@ -2393,6 +2393,18 @@ int of_get_devicepath(struct device_node *partition_node, char **devpath, off_t
 	 */
 	node = partition_node->parent;
 
+	if (of_device_is_compatible(node, "fixed-partitions")) {
+		const char *uuid;
+
+		/* when partuuid is specified short-circuit the search for the cdev */
+		ret = of_property_read_string(partition_node, "partuuid", &uuid);
+		if (!ret) {
+			*devpath = basprintf("/dev/disk/by-partuuid/%s", uuid);
+
+			return 0;
+		}
+	}
+
 	/*
 	 * Respect flash "partitions" subnode. Use parent of parent in this
 	 * case.
-- 
cgit v0.10.2

