From f8074c7dbf7966fd15edd897130825684abe05e9 Mon Sep 17 00:00:00 2001
Message-Id: <f8074c7dbf7966fd15edd897130825684abe05e9.1614713774.git.stefan@agner.ch>
From: Pascal Vizeli <pvizeli@syshack.ch>
Date: Mon, 13 Jan 2020 12:27:06 +0000
Subject: [PATCH 1/5] docker: add AppArmor support

Signed-off-by: Pascal Vizeli <pvizeli@syshack.ch>
Signed-off-by: Stefan Agner <stefan@agner.ch>
---
 package/docker-containerd/docker-containerd.mk | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/package/docker-containerd/docker-containerd.mk b/package/docker-containerd/docker-containerd.mk
index 626889e5f4..7c74da708f 100644
--- a/package/docker-containerd/docker-containerd.mk
+++ b/package/docker-containerd/docker-containerd.mk
@@ -30,6 +30,11 @@ DOCKER_CONTAINERD_DEPENDENCIES += libapparmor
 DOCKER_CONTAINERD_TAGS += apparmor
 endif
 
+ifeq ($(BR2_PACKAGE_LIBAPPARMOR),y)
+DOCKER_CONTAINERD_DEPENDENCIES += libapparmor
+DOCKER_CONTAINERD_TAGS += apparmor
+endif
+
 ifeq ($(BR2_PACKAGE_LIBSECCOMP),y)
 DOCKER_CONTAINERD_DEPENDENCIES += libseccomp host-pkgconf
 DOCKER_CONTAINERD_TAGS += seccomp
-- 
2.30.1

