From 021395083b5829c4fd00bebcdc7d38c2acd7232d Mon Sep 17 00:00:00 2001
From: Pascal Vizeli <pvizeli@syshack.ch>
Date: Wed, 26 Jun 2019 10:00:31 +0000
Subject: [PATCH 1/1] runc: add AppArmor

Signed-off-by: Pascal Vizeli <pvizeli@syshack.ch>
---
 package/runc/runc.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/package/runc/runc.mk b/package/runc/runc.mk
index 55097e5a17..31da4fae58 100644
--- a/package/runc/runc.mk
+++ b/package/runc/runc.mk
@@ -13,7 +13,7 @@ RUNC_WORKSPACE = Godeps/_workspace
 
 RUNC_LDFLAGS = -X main.gitCommit=$(RUNC_VERSION)
 
-RUNC_TAGS = cgo static_build
+RUNC_TAGS = cgo static_build apparmor
 
 ifeq ($(BR2_PACKAGE_LIBSECCOMP),y)
 RUNC_TAGS += seccomp
-- 
2.17.1

