From 51eb8716a560c19656a7eda166ace92ca48888fc Mon Sep 17 00:00:00 2001
From: Pascal Vizeli <pvizeli@syshack.ch>
Date: Sat, 18 Apr 2020 09:03:59 +0000
Subject: [PATCH 1/1] u-boot: fix sysroot stage

Signed-off-by: Pascal Vizeli <pvizeli@syshack.ch>
---
 boot/uboot/uboot.mk | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/boot/uboot/uboot.mk b/boot/uboot/uboot.mk
index 2bfa50779b..9f105c5d26 100644
--- a/boot/uboot/uboot.mk
+++ b/boot/uboot/uboot.mk
@@ -135,6 +135,7 @@ ifeq ($(KERNEL_ARCH),arm64)
 UBOOT_ARCH = arm
 else ifneq ($(filter $(KERNEL_ARCH),i386 x86_64),)
 UBOOT_ARCH = x86
+UBOOT_CC = "--sysroot=$(STAGING_DIR)"
 else
 UBOOT_ARCH = $(KERNEL_ARCH)
 endif
@@ -142,7 +143,7 @@ endif
 UBOOT_MAKE_OPTS += \
 	CROSS_COMPILE="$(TARGET_CROSS)" \
 	ARCH=$(UBOOT_ARCH) \
-	HOSTCC="$(HOSTCC) $(subst -I/,-isystem /,$(subst -I /,-isystem /,$(HOST_CFLAGS)))" \
+	HOSTCC="$(HOSTCC) $(subst -I/,-isystem /,$(subst -I /,-isystem /,$(HOST_CFLAGS))) $(UBOOT_CC)" \
 	HOSTLDFLAGS="$(HOST_LDFLAGS)" \
 	$(call qstrip,$(BR2_TARGET_UBOOT_CUSTOM_MAKEOPTS))
 
-- 
2.17.1

