From a134477d6588a71694d4024360c0b95f305c2aac Mon Sep 17 00:00:00 2001
From: Pascal Vizeli <pvizeli@syshack.ch>
Date: Sat, 31 Mar 2018 16:58:14 +0200
Subject: [PATCH 1/1] NetworkManager: allow to wait on boot

Signed-off-by: Pascal Vizeli <pvizeli@syshack.ch>
---
 package/network-manager/network-manager.mk | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/package/network-manager/network-manager.mk b/package/network-manager/network-manager.mk
index a520aad..67e3767 100644
--- a/package/network-manager/network-manager.mk
+++ b/package/network-manager/network-manager.mk
@@ -100,6 +100,9 @@ define NETWORK_MANAGER_INSTALL_INIT_SYSTEMD
 	ln -sf /usr/lib/systemd/system/NetworkManager.service \
 		$(TARGET_DIR)/etc/systemd/system/multi-user.target.wants/NetworkManager.service
 
+	ln -sf /usr/lib/systemd/system/NetworkManager-wait-online.service \
+		$(TARGET_DIR)/etc/systemd/system/multi-user.target.wants/NetworkManager-wait-online.service
+
 	ln -sf /usr/lib/systemd/system/NetworkManager-dispatcher.service \
 		$(TARGET_DIR)/etc/systemd/system/dbus-org.freedesktop.nm-dispatcher.service
 endef
-- 
2.7.4

