diff --git a/arch/arm/boot/dts/rk3288-tinker.dtsi b/arch/arm/boot/dts/rk3288-tinker.dtsi
index f88c913ff..7f04ccbdf 100644
--- a/arch/arm/boot/dts/rk3288-tinker.dtsi
+++ b/arch/arm/boot/dts/rk3288-tinker.dtsi
@@ -547,3 +547,20 @@
 &gpiomem {
     status = "okay";
 };
+
+&emmc {
+    bus-width = <8>;
+    cap-mmc-highspeed;
+    non-removable;
+    pinctrl-names = "default";
+    pinctrl-0 = <&emmc_clk &emmc_cmd &emmc_pwr &emmc_bus8>;
+    max-frequency = <150000000>;
+    mmc-hs200-1_8v;
+    mmc-ddr-1_8v;
+    status = "okay";
+};
+
+&hdmi {
+    pinctrl-names = "default";
+    pinctrl-0 = <&hdmi_cec_c0>;
+};
