diff --git a/arch/arm/boot/dts/rk3288.dtsi b/arch/arm/boot/dts/rk3288.dtsi
index 58bd91539..9c0dac199 100644
--- a/arch/arm/boot/dts/rk3288.dtsi
+++ b/arch/arm/boot/dts/rk3288.dtsi
@@ -153,6 +153,75 @@
 			opp-hz = /bits/ 64 <1608000000>;
 			opp-microvolt = <1350000>;
 		};
+		opp@1704000000 {
+			opp-hz = /bits/ 64 <1704000000>;
+			opp-microvolt = <1350000>;
+		};
+		opp@1800000000 {
+			opp-hz = /bits/ 64 <1800000000>;
+			opp-microvolt = <1400000>;
+		};
+		/* boot-only frequencies below */
+		opp@1896000000 {
+			opp-hz = /bits/ 64 <1896000000>;
+			opp-microvolt = <1425000>;
+			turbo-mode;
+		};
+		opp@1920000000 {
+			opp-hz = /bits/ 64 <1920000000>;
+			opp-microvolt = <1425000>;
+			turbo-mode;
+		};
+		opp@1992000000 {
+			opp-hz = /bits/ 64 <1992000000>;
+			opp-microvolt = <1450000>;
+			turbo-mode;
+		};
+		opp@2016000000 {
+			opp-hz = /bits/ 64 <2016000000>;
+			opp-microvolt = <1475000>;
+			turbo-mode;
+		};
+		opp@2040000000 {
+			opp-hz = /bits/ 64 <2040000000>;
+			opp-microvolt = <1475000>;
+			turbo-mode;
+		};
+		opp@2064000000 {
+			opp-hz = /bits/ 64 <2064000000>;
+			opp-microvolt = <1475000>;
+			turbo-mode;
+		};
+		opp@2088000000 {
+			opp-hz = /bits/ 64 <2088000000>;
+			opp-microvolt = <1500000>;
+			turbo-mode;
+		};
+		opp@2112000000 {
+			opp-hz = /bits/ 64 <2112000000>;
+			opp-microvolt = <1500000>;
+			turbo-mode;
+		};
+		opp@2136000000 {
+			opp-hz = /bits/ 64 <2136000000>;
+			opp-microvolt = <1500000>;
+			turbo-mode;
+		};
+		opp@2160000000 {
+			opp-hz = /bits/ 64 <2160000000>;
+			opp-microvolt = <1500000>;
+			turbo-mode;
+		};
+		opp@2184000000 {
+			opp-hz = /bits/ 64 <2184000000>;
+			opp-microvolt = <1500000>;
+			turbo-mode;
+		};
+		opp@2208000000 {
+			opp-hz = /bits/ 64 <2208000000>;
+			opp-microvolt = <1500000>;
+			turbo-mode;
+		};
 	};
 
 	amba {
