diff --git a/arch/arm64/boot/dts/amlogic/meson-gxbb-odroidc2.dts b/arch/arm64/boot/dts/amlogic/meson-gxbb-odroidc2.dts
index 0916dcb..cbc03df 100644
--- a/arch/arm64/boot/dts/amlogic/meson-gxbb-odroidc2.dts
+++ b/arch/arm64/boot/dts/amlogic/meson-gxbb-odroidc2.dts
@@ -116,6 +116,32 @@
 			};
 		};
 	};
+
+	spi-gpio {
+		compatible = "spi-gpio";
+		#address-cells = <0x1>;
+		#size-cells = <0x0>;
+		ranges;
+		status = "disabled";
+		sck-gpios  = <&gpio GPIOX_2 0>;
+		miso-gpios = <&gpio GPIOX_4 0>;
+		mosi-gpios = <&gpio GPIOX_7 0>;
+		cs-gpios  = <&gpio GPIOX_3 0
+			     &gpio GPIOX_1 0>;
+		num-chipselects = <2>;
+
+		/* clients */
+		spidev0@0 {
+			compatible = "spidev";
+			reg = <0>;
+			spi-max-frequency = <500000>;
+		};
+		spidev0@1 {
+			compatible = "spidev";
+			reg = <1>;
+			spi-max-frequency = <500000>;
+		};
+	};
 
 	sound {
 		compatible = "simple-audio-card";
