From 1529062f72cee8061b4733e1148357d227639900 Mon Sep 17 00:00:00 2001
From: Pascal Vizeli <pvizeli@syshack.ch>
Date: Mon, 11 Jun 2018 11:09:10 +0000
Subject: [PATCH 1/1] Support HassOS bootstate partition

Signed-off-by: Pascal Vizeli <pvizeli@syshack.ch>
---
 env/mmc.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/env/mmc.c b/env/mmc.c
index 5e3da6dca7..75c06899d1 100644
--- a/env/mmc.c
+++ b/env/mmc.c
@@ -18,6 +18,10 @@
 #include <search.h>
 #include <errno.h>
 
+#define CONFIG_SYS_MMC_ENV_DEV 0
+#define CONFIG_ENV_OFFSET 0x25100000
+#define CONFIG_ENV_SIZE 0x400000
+
 #define __STR(X) #X
 #define STR(X) __STR(X)
 
-- 
2.17.1

