From 17ddce76cfe4d76b0b55b6dcd6d2a072e760f6cf Mon Sep 17 00:00:00 2001
From: Pascal Vizeli <pvizeli@syshack.ch>
Date: Thu, 23 Apr 2020 11:07:08 +0000
Subject: [PATCH 1/1] raspberrypi: disable GPIO fixups p2

Signed-off-by: Pascal Vizeli <pvizeli@syshack.ch>
---
 drivers/serial/Makefile | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/drivers/serial/Makefile b/drivers/serial/Makefile
index e4a92bbbb7..ee89aa4f08 100644
--- a/drivers/serial/Makefile
+++ b/drivers/serial/Makefile
@@ -21,13 +21,13 @@ endif
 
 endif
 
-ifdef CONFIG_DM_SERIAL
-obj-$(CONFIG_PL01X_SERIAL) += serial_pl01x.o
-else
-obj-$(CONFIG_PL010_SERIAL) += serial_pl01x.o
-obj-$(CONFIG_PL011_SERIAL) += serial_pl01x.o
-obj-$(CONFIG_SYS_NS16550_SERIAL) += serial_ns16550.o
-endif
+#ifdef CONFIG_DM_SERIAL
+#obj-$(CONFIG_PL01X_SERIAL) += serial_pl01x.o
+#else
+#obj-$(CONFIG_PL010_SERIAL) += serial_pl01x.o
+#obj-$(CONFIG_PL011_SERIAL) += serial_pl01x.o
+#obj-$(CONFIG_SYS_NS16550_SERIAL) += serial_ns16550.o
+#endif
 
 obj-$(CONFIG_ALTERA_UART) += altera_uart.o
 obj-$(CONFIG_ALTERA_JTAG_UART) += altera_jtag_uart.o
@@ -60,8 +60,8 @@ obj-$(CONFIG_UNIPHIER_SERIAL) += serial_uniphier.o
 obj-$(CONFIG_STM32_SERIAL) += serial_stm32.o
 obj-$(CONFIG_STI_ASC_SERIAL) += serial_sti_asc.o
 obj-$(CONFIG_PIC32_SERIAL) += serial_pic32.o
-obj-$(CONFIG_BCM283X_MU_SERIAL) += serial_bcm283x_mu.o
-obj-$(CONFIG_BCM283X_PL011_SERIAL) += serial_bcm283x_pl011.o
+#obj-$(CONFIG_BCM283X_MU_SERIAL) += serial_bcm283x_mu.o
+#obj-$(CONFIG_BCM283X_PL011_SERIAL) += serial_bcm283x_pl011.o
 obj-$(CONFIG_MSM_SERIAL) += serial_msm.o
 obj-$(CONFIG_MVEBU_A3700_UART) += serial_mvebu_a3700.o
 obj-$(CONFIG_MPC8XX_CONS) += serial_mpc8xx.o
-- 
2.17.1

