From defc964eaa9f4ccc9cd1568ade4e4d81f35b9e07 Mon Sep 17 00:00:00 2001
From: Pascal Vizeli <pvizeli@syshack.ch>
Date: Fri, 24 Apr 2020 11:37:55 +0000
Subject: [PATCH] pmic: enable LDO2 vcc33_mipi at bootup

Signed-off-by: Pascal Vizeli <pvizeli@syshack.ch>
---
 board/rockchip/tinker_rk3288/tinker-rk3288.c | 37 ++++++++++++++++++++
 1 file changed, 37 insertions(+)

diff --git a/board/rockchip/tinker_rk3288/tinker-rk3288.c b/board/rockchip/tinker_rk3288/tinker-rk3288.c
index f85209c649..6fa1bb0c66 100644
--- a/board/rockchip/tinker_rk3288/tinker-rk3288.c
+++ b/board/rockchip/tinker_rk3288/tinker-rk3288.c
@@ -13,6 +13,7 @@
 #include <netdev.h>
 #include <asm/arch-rockchip/bootrom.h>
 #include <asm/io.h>
+#include <power/regulator.h>
 
 static int get_ethaddr_from_eeprom(u8 *addr)
 {
@@ -48,3 +49,39 @@ int mmc_get_env_dev(void)
 
 	return 1;
 }
+
+
+#ifdef CONFIG_DM_PMIC
+static int rockchip_set_regulator_on(const char *name, uint uv)
+{
+	struct udevice *dev;
+	int ret;
+
+	ret = regulator_get_by_platname(name, &dev);
+	if (ret) {
+		debug("%s: Cannot find regulator %s\n", __func__, name);
+		return ret;
+	}
+	ret = regulator_set_value(dev, uv);
+	if (ret) {
+		debug("%s: Cannot set regulator %s\n", __func__, name);
+		return ret;
+	}
+	ret = regulator_set_enable(dev, 1);
+	if (ret) {
+		debug("%s: Cannot enable regulator %s\n", __func__, name);
+		return ret;
+	}
+
+	return 0;
+}
+
+int power_init_board(void)
+{
+	int ret = rockchip_set_regulator_on("vcc33_mipi", 3300000);
+	if (ret)
+		return ret;
+
+	return 0;
+}
+#endif
-- 
2.43.0

