From cb583beba1acba55112fbb4636c1c73d9ae691a0 Mon Sep 17 00:00:00 2001
From: Johan Oudinet <johan.oudinet@gmail.com>
Date: Fri, 20 Nov 2020 00:33:30 +0100
Subject: [PATCH] Makefile.in: do not download or compile dependencies

Signed-off-by: Johan Oudinet <johan.oudinet@gmail.com>
---
 Makefile.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index abd64835..06f876bb 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -97,7 +97,7 @@ ifneq ($(INSTALLGROUP),)
   G_USER=-g $(INSTALLGROUP)
 endif
 
-all: deps src
+all: src
 
 deps: deps/.got
 
@@ -111,7 +111,7 @@ deps/.built: deps/.got
 	$(REBAR) configure-deps
 	$(REBAR) compile && :> deps/.built
 
-src: deps/.built
+src:
 	$(REBAR) skip_deps=true compile
 
 update:
-- 
2.29.2

