--- ./dts/pivccu-raspberrypi.dts.orig	2018-12-22 21:01:28.756992536 +0100
+++ ./dts/pivccu-raspberrypi.dts	2018-12-26 22:56:26.807677174 +0100
@@ -7,10 +7,10 @@
     target = <&uart0>;
     __overlay__ {
       compatible = "pivccu,pl011";
-      pivccu,reset_pin = <&gpio 18 0>;
-      pivccu,red_pin = <&gpio 16 0>;
-      pivccu,green_pin = <&gpio 20 0>;
-      pivccu,blue_pin = <&gpio 21 0>;
+      pivccu,reset_pin = <0>;
+      pivccu,red_pin = <0>;
+      pivccu,green_pin = <0>;
+      pivccu,blue_pin = <0>;
     };
   };
 };
