--- ./dts/pivccu-raspberrypi.dts.orig	2019-07-04 13:54:53.271472462 +0200
+++ ./dts/pivccu-raspberrypi.dts	2021-03-18 22:10:52.040233559 +0100
@@ -23,10 +23,10 @@
       status = "okay";
 
       rx8130@32 {
-        compatible = "epson,rx8130-legacy";
+        compatible = "epson,rx8130";
         reg = <0x32>;
         status = "okay";
-	enable-external-capacitor;
+        aux-voltage-chargeable = <1>;
       };
     };
   };
