diff --git a/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-n2.dts b/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-n2.dts
index 93ba39044eb3..e2f727ffb731 100644
--- a/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-n2.dtsi
+++ b/arch/arm64/boot/dts/amlogic/meson-g12b-odroid-n2.dtsi
@@ -24,6 +24,7 @@ aliases {
 		serial1 = &uart_A;
 		i2c0 = &i2c2;
 		i2c1 = &i2c3;
+		rtc1 = &vrtc;
 	};
 
 	dioo2133: audio-amplifier-0 {
@@ -632,6 +633,11 @@ &i2c3 {
 
 	/* default 100k */
 	clock-frequency = <100000>;
+	
+	rtc@51 {
+		compatible = "nxp,pcf8563";
+		reg = <0x51>;
+	};
 };
 
 &ir {
