From 3af12d8d0f749751e487955c24260049e7a8eec6 Mon Sep 17 00:00:00 2001
From: Kamel Bouhara <kamel.bouhara@bootlin.com>
Date: Wed, 13 May 2020 16:58:22 +0200
Subject: [PATCH] fix syntax issue while checking ldflags

Signed-off-by: Kamel Bouhara <kamel.bouhara@bootlin.com>
---
 configure | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure b/configure
index def5883..2d783f6 100755
--- a/configure
+++ b/configure
@@ -17574,7 +17574,7 @@ libmysqlclient_paths="none /usr/lib/mysql /usr/lib64/mysql"
 
 for path in $libmysqlclient_paths; do
 
-if test x"path" != xnone; then
+if test x"$path" != xnone; then
   save_LDFLAGS="$LDFLAGS"
   LDFLAGS="$LDFLAGS -L$path"
 fi
@@ -17753,7 +17753,7 @@ if test x"$libmysqlclient_found" = xyes; then
   break;
 fi
 
-if test x"path" != xnone; then
+if test x"$path" != xnone; then
   LDFLAGS="$save_LDFLAGS"
 fi
 
-- 
2.26.2

