From 78ceb52ff4e5d4dbe003651b2193979114152763 Mon Sep 17 00:00:00 2001
From: Pascal Vizeli <pvizeli@syshack.ch>
Date: Mon, 30 Apr 2018 23:40:27 +0200
Subject: [PATCH 1/1] Fix permission

---
 parser/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/parser/Makefile b/parser/Makefile
index b18cfe4..7b7b519 100644
--- a/parser/Makefile
+++ b/parser/Makefile
@@ -383,7 +383,7 @@ install-indep: indep
 install-systemd:
 	install -m 755 -d $(SYSTEMD_UNIT_DIR)
 	install -m 644 apparmor.service $(SYSTEMD_UNIT_DIR)
-	install -m 644 apparmor.systemd $(APPARMOR_BIN_PREFIX)
+	install -m 755 apparmor.systemd $(APPARMOR_BIN_PREFIX)
 	install -m 755 -d $(DESTDIR)/sbin
 	install -m 755 aa-teardown $(DESTDIR)/sbin
 
-- 
2.7.4

