
--- b/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c	2018-04-12 01:40:29.029325568 +0200
+++ a/drivers/net/ethernet/stmicro/stmmac/stmmac_main.c	2018-04-14 01:12:07.301469106 +0200
@@ -364,15 +364,9 @@
 bool stmmac_eee_init(struct stmmac_priv *priv)
 {
 	struct net_device *ndev = priv->dev;
-	int interface = priv->plat->interface;
 	unsigned long flags;
 	bool ret = false;
 
-	if ((interface != PHY_INTERFACE_MODE_MII) &&
-	    (interface != PHY_INTERFACE_MODE_GMII) &&
-	    !phy_interface_mode_is_rgmii(interface))
-		goto out;
-
 	/* Using PCS we cannot dial with the phy registers at this stage
 	 * so we do not support extra feature like EEE.
 	 */
