--- a/arch/arm64/boot/dts/amlogic/meson-gxbb-odroidc2.dts	2018-05-01 01:15:01.378785083 +0200
+++ b/arch/arm64/boot/dts/amlogic/meson-gxbb-odroidc2.dts	2018-05-02 00:16:22.834304176 +0200
@@ -185,7 +185,7 @@
 
 	sound {
 		compatible = "simple-audio-card";
-		simple-audio-card,name = "meson-gx-preview";
+		simple-audio-card,name = "meson-gxbb-audio";
 		status = "okay";
 
 		simple-audio-card,dai-link@0 {
