diff -ur a/drivers/gpu/drm/meson/meson_vclk.c b/drivers/gpu/drm/meson/meson_vclk.c
--- a/drivers/gpu/drm/meson/meson_vclk.c	2018-06-02 20:21:48.000000000 +0200
+++ b/drivers/gpu/drm/meson/meson_vclk.c	2018-06-02 20:16:44.008061996 +0200
@@ -428,9 +428,9 @@
 		.vclk_div = 2,
 	},
 	[MESON_VCLK_HDMI_32000] = {
-		.pll_base_freq = 2560000,
+		.pll_base_freq = 5120000,
 		.pll_od1 = 4,
-		.pll_od2 = 2,
+		.pll_od2 = 4,
 		.pll_od3 = 1,
 		.vid_pll_div = VID_PLL_DIV_5,
 		.vclk_div = 2,
@@ -520,27 +520,6 @@
 
 	if (meson_vpu_is_compatible(priv, "amlogic,meson-gxbb-vpu")) {
 		switch (base) {
-		case 2560000:
-			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL, 0x58000235);
-			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL2, 0x00000000);
-			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL3, 0x0d5c5091);
-			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL4, 0x801da72c);
-			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL5, 0x71486980);
-			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL6, 0x00000e55);
-
-			/* Enable and unreset */
-			regmap_update_bits(priv->hhi, HHI_HDMI_PLL_CNTL,
-						0x7 << 28, 0x4 << 28);
-
-			/* Poll for lock bit */
-			regmap_read_poll_timeout(priv->hhi, HHI_HDMI_PLL_CNTL,
-					val, (val & HDMI_PLL_LOCK), 10, 0);
-
-			/* div_frac */
-			regmap_update_bits(priv->hhi, HHI_HDMI_PLL_CNTL2,
-						0xFFFF,  0x4555);
-			break;
-
 		case 2700000:
 			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL, 0x58000238);
 			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL2, 0x00000000);
@@ -722,6 +701,27 @@
 					val, (val & HDMI_PLL_LOCK), 10, 0);
 			break;
 
+		case 5120000:
+			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL, 0x5800026a);
+			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL2, 0x00000000);
+			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL3, 0x135c5091);
+			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL4, 0x801da72c);
+			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL5, 0x71486980);
+			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL6, 0x00000e55);
+
+			/* unreset */
+			regmap_update_bits(priv->hhi, HHI_HDMI_PLL_CNTL,
+						BIT(28), 0);
+
+			/* Poll for lock bit */
+			regmap_read_poll_timeout(priv->hhi, HHI_HDMI_PLL_CNTL,
+					val, (val & HDMI_PLL_LOCK), 10, 0);
+
+			/* div_frac */
+			regmap_update_bits(priv->hhi, HHI_HDMI_PLL_CNTL2,
+						0xFFFF,  0x4aab);
+			break;
+
 		case 5940000:
 			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL, 0x5800027b);
 			regmap_update_bits(priv->hhi, HHI_HDMI_PLL_CNTL2,
@@ -760,15 +760,6 @@
 	} else if (meson_vpu_is_compatible(priv, "amlogic,meson-gxm-vpu") ||
 		   meson_vpu_is_compatible(priv, "amlogic,meson-gxl-vpu")) {
 		switch (base) {
-		case 2560000:
-			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL, 0x4000026a);
-			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL2, 0x800cb2ab);
-			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL3, 0x860f30c4);
-			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL4, 0x0c8e0000);
-			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL5, 0x001fa729);
-			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL6, 0x01a31500);
-			break;
-
 		case 2700000:
 			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL, 0x40000270);
 			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL2, 0x800cb200);
@@ -847,6 +838,15 @@
 			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL3, 0x860f30c4);
 			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL4, 0x0c8e0000);
 			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL5, 0x001fa729);
+			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL6, 0x01a31500);
+			break;
+
+		case 5120000:
+			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL, 0x400002d5);
+			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL2, 0x800cb155);
+			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL3, 0x860f30c4);
+			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL4, 0x0c8e0000);
+			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL5, 0x001fa729);
 			regmap_write(priv->hhi, HHI_HDMI_PLL_CNTL6, 0x01a31500);
 			break;
 
diff -ur a/drivers/gpu/drm/meson/meson_venc.c b/drivers/gpu/drm/meson/meson_venc.c
--- a/drivers/gpu/drm/meson/meson_venc.c	2018-06-02 16:39:27.777402009 +0200
+++ b/drivers/gpu/drm/meson/meson_venc.c	2018-06-02 20:23:23.744646670 +0200
@@ -713,6 +713,12 @@
 		  968, 1056, 0, 600, 601, 605, 628, 0,
 		  DRM_MODE_FLAG_PHSYNC | DRM_MODE_FLAG_PVSYNC) },
 	},
+	/* 1024x600@43Hz */
+	{
+		{ DRM_MODE("1024x600", DRM_MODE_TYPE_DRIVER, 32000, 1024,
+		  1064, 1112, 1152, 0, 600, 613, 616, 645, 0,
+		  DRM_MODE_FLAG_NHSYNC | DRM_MODE_FLAG_NVSYNC) },
+	},
 	/* 1024x768@60Hz */
 	{
 		{ DRM_MODE("1024x768", DRM_MODE_TYPE_DRIVER, 65000, 1024,
